### PR TITLE
Add check command

### PR DESCRIPTION
**Pull request by uberhacker**
Tested and working.  Version check inspired by [version.rb](https://github.com/Zopolis4/chromebrew/blob/vcheck/tools/version.rb).  Fixed a bug with `crew search -v` and provided better error handling for missing or invalid command arguments.

### CODE REVIEW

**Comment by Zopolis4** [comment]
`version.rb` is unfinished (still should be mostly functional) and this is not how I would have implemented things-- happy to let this get merged and used until I sort things out, though.

**Comment by uberhacker** [comment]
> `version.rb` is unfinished (still should be mostly functional) and this is not how I would have implemented things-- happy to let this get merged and used until I sort things out, though.

How would you implement this then?  What do you need to sort out?

**Comment by Zopolis4** [comment]
I probably wouldn't have a check command at all-- both of these files are in `tools` for a reason, and especially `prop_test`, which is already part of our CI. The version of `version.rb` you have is missing the libversion integration (the one on my `vcheck` branch has that), and I haven't finished working on properly excluding packages or handling missing packages. 

Again, though, I'm happy for this to be merged. I'm not going to hold up a feature from being merged now just because I plan to improve it in the future.

File: tools/version.rb [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  'gtksourceview_3',
+  'gtksourceview_4',
+  'gtksourceview_5',
+  'imagemagick6',
+  'imagemagick7',
+  'jdk11',
+  'jdk17',
+  'jdk18',
+  'jdk8',
+  'js102',
```
Comment (satmandu) [inline tools/version.rb:118]: Add js115?

**Review by satmandu — APPROVED** [review]

File: tools/version.rb (hunk 1 above)
Comment (uberhacker) [reply tools/version.rb]: Replaced list with `no_upstream_update` property.

**Comment by Zopolis4** [comment]
These changes to prop_test are really slowing down CI (see some of the runs on #9514)-- I'm not entirely sure why, and it only seems to be happening sometimes.

**Comment by uberhacker** [comment]
> These changes to prop_test are really slowing down CI (see some of the runs on #9514)-- I'm not entirely sure why, and it only seems to be happening sometimes.

There are over 2k packages now.  The last run took only a min so not too bad...faster than I would have expected.

**Pull request merged**